structure/set/unionfind.go

```diff
@@ -0,0 +1,63 @@
+package set
+
+// https://en.wikipedia.org/wiki/Disjoint-set_data_structure
+
+type Value interface{}
+
+type node struct {
+	rank  int
+	value Value
```

Comment:
The usual way is to refer to this as `parent` rather than `value`. Keep it consistent with all the documentation that is available online for people to refer to. Even the Wikipedia article you refer to calls this parent. And instead of making this `Value` interface you should consider it to be a node. All your implementations would be simplified with that.